### ISSUE TITLE
push vs PUSH

### ISSUE THREAD

**Issue by dpwrussell** [FormidableLabs/redux-little-router#196]
As there were no action-creators in RLR, I was using `PUSH` type like so.

```js
  dispatch({
    type: PUSH,
    payload: '/results?q=a'
  });
```

I just upgraded to 14.0.0-0 to test an upgrade and my middleware stopped working because there seems to have been a switch from parsing the query parameters in the middleware, to doing it in the action-creator I think.

If I do the above now, I will get an action with the correct `pathname` and `search`, but with nothing in the `query` object.

Given that using `PUSH` was necessary until recently, I think this deserves to be prominent in the action-creator section of the documentation at the least. It actually seems like potentially more of a bug as I don't see what advantage is gained by moving some of the router behaviour.

Minimal demo example: https://github.com/dpwrussell/RLRQueryIssueDemo

**Comment by tptee** [comment]
Hey @dpwrussell ,

I don't think we ever parsed anything in the middleware–a lot of that happened in `Link` previously. I moved the parsing to the action creators for a few reasons:

- We want the action creators to do anything `Link` can: parsing queries, normalizing string/object "hrefs", requesting query persistence, etc.
- We had to move `persistQuery` handling to the reducer because the action creators don't have access to the current state (and we don't want to require `redux-thunk` for this)

I'll add a note in the action creator section discouraging the use of the bare action types. I might consider removing them from the export list as well so they're not public API anymore 😄
